### ISSUE TITLE
Please allow Firefox for Android

### ISSUE THREAD

**Issue by msxfm** [RequestPolicyContinued/requestpolicy#416]
<a href="https://github.com/concat22"><img src="https://avatars.githubusercontent.com/u/6267930?" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [concat22](https://github.com/concat22)**
_Monday Jan 06, 2014 at 07:32 GMT_
_Originally opened as https://github.com/RequestPolicy/requestpolicy/issues/416_

---

I can't install requestpolicy on Android version of Firefox.
It's important to block 3rd-party codes(such as js,css,and bad ads) to protect privacy and time.

**Comment by martintietz** [comment]
+1

**Comment by myrdd** [comment]
+1 from me, too ;) Sadly, the GUI is not flexible enough yet, and RP without an GUI is not very useful… If anyone wants to take a look, it's especilly `overlay.js` and `menu.js` that have to be modified for *Fx on Android*.

**Comment by Cattze2015** [comment]
+1.

GUI is not so important, just adding a menu will do a trick.
For example: https://noscript.net/nsa/

**Comment by ivan-kolmychek** [comment]
@myrdd do you mean RP GUI or Firefox one?

**Comment by myrdd** [comment]
> GUI is not so important, just adding a menu will do a trick.

true, but it would be cool if it was that easy. Okay, I didn't try it so far, but by experience the details make it complicated.

If anyone wants to work on it, I'd appreciate that! And of course I'd help as much as I can. But I for myself will work on the desktop version for the next few months, sorry.

In any case, I think it would be good to aim an Android version still this year. Maybe I can work on it in summer.

> do you mean RP GUI or Firefox one?

In my comment above I meant RP's GUI. Maybe it's not that difficult to get a menu for RP, but I guess it would be „hacky“. If you look at the current state of [`menu.js`](https://github.com/RequestPolicyContinued/requestpolicy/blob/581e2b8547917df903a8d4664728f1c469a5109e/src/content/ui/menu.js) you maybe understand what I mean.

**Comment by Gomat** [comment]
I don't have the chops to help but offer muchos kudos to anyone who can pitch in.

**Comment by qasmoke** [comment]
It's 2016, and still +1.

**Comment by jdgalt** [comment]
+1

**Comment by CorvusCorax** [comment]
+1

**Comment by ivan-kolmychek** [comment]
For those who wants to put `+1`: please, just submit a "reaction" to the main post and click subscribe button on the right instead of doing `+1`, as it sens the message to all the participants. We all would want to see more relevant messages on this matter, not just `+1` once in a while. ;)

Thank you for understanding. :)

**Comment by myrdd** [comment]
@ivan-kolmychek exactly! :)

To give an update: Regarding the dependencies of this issue, …

> the GUI is not flexible enough yet, [...] especilly `overlay.js` and `menu.js`

… I've got quite some work on my local disk, making any changes to the menu easier. Essentially, it's a `BaseMenu` class, which separates any GUI-independant code from the actual graphical interface (`menu.js`). The first issue benefiting from this new class will be the [re-introduction of the strictness](https://github.com/RequestPolicyContinued/requestpolicy/issues/474), and many closely related issues (https://github.com/RequestPolicyContinued/requestpolicy/issues/619, https://github.com/RequestPolicyContinued/requestpolicy/issues/473, https://github.com/RequestPolicyContinued/requestpolicy/issues/524, https://github.com/RequestPolicyContinued/requestpolicy/issues/724).
